APIs/API_DrawPanel/API_DrawPanel.py

```diff
@@ -0,0 +1,371 @@
+        self.counts = {}
+        for effect in self.status_effects:
+            if effect.name not in self.counts:
+                self.counts[effect.name] = (effect, 0, 0, None)
+            _, stacks, duration, color = self.counts[effect.name]
+            stacks += 1
+            duration = max(duration, effect.turns_left)
+
+            self.counts[effect.name] = (effect, stacks, duration, effect.get_tooltip_color().to_tup())
+
+    def get_height(self):
+        return self.linesize * (2 + len(self.counts))
+
+    def should_draw(self):
+        return self.status_effects
+
+    def draw(self, cur_x, cur_y):
+        # Buffs
+        
+        cur_y += self.linesize
+        self.draw_string("Status Effects:", self.character_display, cur_x, cur_y, (255, 255, 255))
+        cur_y += self.linesize
+        for buff_name, (buff, stacks, duration, color) in self.counts.items():
+            fmt = buff_name
+            if stacks > 1:
+                fmt += ' x%d' % stacks
+            if duration:
+                fmt += ' (%d)' % duration
+            self.draw_string(fmt, self.character_display, cur_x, cur_y, color, mouse_content=buff)
+            cur_y += self.linesize
+
+class PanelSkills(PanelItem):
+    def setup(self):
+        self.skills = [b for b in self.game.p1.buffs if b.buff_type == RiftWizard.BUFF_TYPE_PASSIVE and not b.prereq]
+    
+    def get_height(self): # This is scuffed. Might want to just make draw return the height.....
+        height = self.linesize * 2
+        cur_x = self.border_margin
+        skill_x_max = self.character_display.get_width() - self.border_margin - 16
+        for skill in self.skills:
+            cur_x += 18
+            if cur_x > skill_x_max:
+                cur_x = self.border_margin
+                height += self.linesize
+        return height
+
+    def should_draw(self):
+        return self.skills
+
+    def draw(self, cur_x, cur_y):
+        cur_y += self.linesize
+
+        self.draw_string("Skills:", self.character_display, cur_x, cur_y)
+        cur_y += self.linesize
+
+        skill_x_max = self.character_display.get_width() - self.border_margin - 16
+        for skill in self.skills:
+            self.draw_spell_icon(skill, self.character_display, cur_x, cur_y)
+            cur_x += 18
+            if cur_x > skill_x_max:
+                cur_x = self.border_margin
+                cur_y += self.linesize
+
+class PanelSeperator(PanelItem):
+
+    def __init__(self, order, lines=1):
+        super().__init__(order)
+        self.lines = lines
+
+    def get_height(self):
+        return self.linesize * self.lines
+
+PANEL_HP = PanelHP(0)
+PANEL_SHIELDS = PanelShield(100)
+PANEL_XP = PanelXP(200)
+PANEL_REALM = PanelRealm(300)
+PANEL_SPELLS = PanelSpells(400)
+PANEL_ITEMS = PanelItems(500)
+PANEL_BUFFS = PanelBuffs(600)
+PANEL_SKILLS = PanelSkills(700)
+
+character_panels: List[PanelItem] = [
```

Comment:
I love that order is defined in the Panel constructor rather than the order of this list.